src/pytorch_lightning/utilities/apply_func.py

```diff
@@ -40,6 +41,10 @@
 
 
 _CPU_DEVICES = ("cpu", torch.device("cpu"))
+if _TORCH_GREATER_EQUAL_1_12 and torch.backends.mps.is_available() and platform.processor() in ("arm", "arm64"):
```

Comment:
You can import and use the same check in `src/pytorch_lightning/accelerators/mps.py`

Reply:
Thanks for the review.

Good point. I'm fixing this now. However, the simple approach of importing `MPSAccelerator.is_available()` or `_MPS_AVAILABLE` does not work because import `pytorch_lightning.accelerators.mps` creates a circular import:
```ImportError: cannot import name '_TPU_AVAILABLE' from partially initialized module 'pytorch_lightning.utilities'`. 
I tried to fix this by moving `_MPS_AVAILABLE`. I guess  `pytorch_lightning.utilities.imports`  is the right place for this then, but I'm not sure. What do you think?

Reply:
I was able to resolve the circular dependency in https://github.com/Lightning-AI/lightning/pull/14368/commits/02220a711bbfb693a5b295a2304d9cd3033b8875

We prefer keeping the availability checks in the files they are used: https://github.com/Lightning-AI/lightning/issues/11826. The ones in imports just haven't been updated.